src/protobuf.zig

```diff
@@ -363,17 +430,23 @@ fn internal_pb_encode(pb: *ArrayList(u8), data: anytype) !void {
     inline for (field_list) |field| {
         if (@typeInfo(field.type) == .Optional) {
             if (@field(data, field.name)) |value| {
-                try append(pb, @field(data_type._desc_table, field.name), @TypeOf(value), value);
+                try append(pb, @field(data_type._desc_table, field.name), value);
             }
         } else {
             switch (@field(data_type._desc_table, field.name).ftype) {
-                .List => if (@field(data, field.name).items.len != 0) {
-                    try append(pb, @field(data_type._desc_table, field.name), @TypeOf(@field(data, field.name)), @field(data, field.name));
+                .List, .PackedList => if (@field(data, field.name).items.len != 0) {
+                    try append(pb, @field(data_type._desc_table, field.name), @field(data, field.name));
                 },
                 .Map => if (@field(data, field.name).count() != 0) {
-                    try append(pb, @field(data_type._desc_table, field.name), @TypeOf(@field(data, field.name)), @field(data, field.name));
+                    try append(pb, @field(data_type._desc_table, field.name), @field(data, field.name));
+                },
+                .Varint, .FixedInt => if (@as(u64, @field(data, field.name)) != 0) {
+                    try append(pb, @field(data_type._desc_table, field.name), @field(data, field.name));
                 },
```

Comment:
This is slightly off the spec (right now). The expected behavior is something like

```zig
// In this case, we are assuming that the default value of each
// field is 0 to omit it on the encoded message.
if (data.field != null and data.field != meta(data.field).default_value) {
  write_field(data.field)
}
```



In the totality of my changes, I assume required fields are unexistent. and in some cases, it may only be a hint for the typesystem, a help to the user to prevent many `.?` everywhere.